### ISSUE TITLE
[@angular-eslint/no-implicit-take-until-destroyed] false positive

### ISSUE THREAD

**Issue by splincode** [angular-eslint/angular-eslint#3074]
I have

https://github.com/taiga-family/taiga-ui/blob/main/projects/demo/src/pages/app/utils/viewport-scroller.service.ts

```ts
 constructor() {
        inject(TUI_DOC_PAGE_LOADED)
            .pipe(
                filter(Boolean),
                switchMap(() => this.scroll$),
                takeUntilDestroyed(),
            )
            .subscribe((element) => element.scrollIntoView());
    }
```

But

```
projects/demo/src/pages/app/utils/viewport-scroller.service.ts
  20:17  error  `takeUntilDestroyed()` must be called with an explicit `DestroyRef` parameter when used outside of an injection context. See more at https://angular.dev/guide/di/dependency-injection-context and https://angular.dev/ecosystem/rxjs-interop/take-until-destroyed  @angular-eslint/no-implicit-take-until-destroyed

✖ 1 problem (1 error, 0 warnings)
```

This is normal behavior for [Angular](https://angular.dev/ecosystem/rxjs-interop/take-until-destroyed), and you don't need to use destroyRef.

**Versions**

| package                         | version |
| ------------------------------- | ------- |
| `@angular-eslint/eslint-plugin` | `21.4.0` |
| `@typescript-eslint/parser`     | `8.60.0` |
| `ESLint`                        | `9.39.4` |
| `node`                          | `26.2.0` |

**Comment by JamesHenry** [comment]
Hi @splincode, having looked into this, at the time this was reported/experienced this class did not have `@Injectable` so this was intended behaviour.

I would not change the rule to allow undecorated constructors.

- This is a problem rule. Missing a real `NG0203` is worse than asking for `@Injectable()`.
- `@Injectable()` is the idiomatic Angular fix and matches what you later did.
- We cannot statically know whether a plain class is DI-constructed.
- The current tests treat this as expected.
